app/services/oidc_jwt_token_validate.rb

```diff
@@ -0,0 +1,51 @@
+class OidcJwtTokenValidate
+  attr_reader :decoded_token
+  delegate :logger, to: :Rails
+
+  def initialize(jwt_token)
+    @jwt_token = jwt_token
+  end
+
+  def valid_jwt?
+    valid_signature? && type_bearer? && valid_aud?
+  rescue JWT::InvalidAudError
+    logger.error "JWT SSO: Invalid token."
```

Comment:
I'd chosse `OidcJwtToken SSO:` as a prefix here to make sure we're using this SSO method.